packages/contracts/src/framework/plugin/setup/PluginSetup.sol

```diff
@@ -14,6 +14,13 @@ import {IPluginSetup} from "./IPluginSetup.sol";
 /// @author Aragon Association - 2022-2023
 /// @notice An abstract contract that developers have to inherit from to write the setup of a plugin.
 abstract contract PluginSetup is ERC165, IPluginSetup {
+    /// @notice The implementation contract.
+    address public immutable implementation;
```

Comment:
Doesn't this only make sense for upgradeable contracts? Why not having it on `IPluginSetup`?

Reply:
> Doesn't this only make sense for upgradeable contracts? 

No, `PluginCloneable` contracts are proxies as well and need it too.

> Why not having it on `IPluginSetup`?

Variables, immutables, and constants cannot be put in an interface.